pkg/network/tracer/testutil/testdns/test_dns_server.go

```diff
@@ -20,20 +21,33 @@ import (
 var globalServer *server
 var globalServerError error
 var serverOnce sync.Once
+var non53ServerOnce sync.Once
+
+const localhostAddr = "127.0.0.1"
+
+func GetServerIPPort53(t *testing.T) net.IP {
+	return GetServerIP(t, 53)
+}
 
 // GetServerIP returns the IP address of the test DNS server. The test DNS server returns canned responses for several
 // known domains that are used in integration tests.
 //
 // see server#start to see which domains are handled.
-func GetServerIP(t *testing.T) net.IP {
+func GetServerIP(t *testing.T, port int) net.IP {
+	if port != 53 {
+		non53ServerOnce.Do(func() {
+			globalServer.Start("tcp", port)
+			globalServer.Start("udp", port)
+		})
+	}
 	serverOnce.Do(func() {
 		globalServer, globalServerError = newServer()
-		globalServer.Start("tcp")
-		globalServer.Start("udp")
+		globalServer.Start("tcp", port)
+		globalServer.Start("udp", port)
 	})
 
 	require.NoError(t, globalServerError)
-	return net.ParseIP("127.0.0.1")
+	return net.ParseIP(localhostAddr)
 }
```

Comment:
I think this could be cleaner if GetServerIP had it's original signature (taking a `*testing.T`) and started both the non-port-53 server and the port-53 server.

Reply:
yeah that's fine... I think the only thing there is that the non-53 server will be started always, when it's only used in one test